src/status_im/contexts/chat/messenger/composer/link_preview/events.cljs

```diff
@@ -82,22 +82,45 @@
 
 (rf/defn unfurl-parsed-urls-success
   {:events [:link-preview/unfurl-parsed-urls-success]}
-  [{:keys [db]} request-id {new-previews :linkPreviews}]
+  [{:keys [db]} request-id {new-previews :linkPreviews status-link-previews :statusLinkPreviews}]
   (when (= request-id (get-in db [:chat/link-previews :request-id]))
-    (let [new-previews         (map data-store.messages/<-link-preview-rpc new-previews)
-          curr-previews        (get-in db [:chat/link-previews :unfurled])
-          indexed-new-previews (utils.collection/index-by :url new-previews)]
+    (let [new-previews             (map data-store.messages/<-link-preview-rpc new-previews)
+          new-status-link-previews (map data-store.messages/<-status-link-previews-rpc
+                                        status-link-previews)
+          merged-preview-types     (-> (map
```

Comment:
Consider extracting `merged-preview-types` to a separate function outside the event handler. In general, Clojure code reads better without long let bindings. In this case, it would also simplify reasoning about what the function needs. When it's inlined in the let binding, the dev is left to read line by line to find if it's closing over any values. It's also harder to play in the REPL with inlined functions.